bin/check-temperature.rb

```diff
@@ -0,0 +1,88 @@
+#! /usr/bin/env ruby
+#
+#   check-temperature
+#
+# DESCRIPTION:
+#
+# OUTPUT:
+#   plain text
+#
+# PLATFORMS:
+#   Linux
+#
+# DEPENDENCIES:
+#   gem: sensu-plugin
+#   gem: socket
+#   lm-sensors
+#
+# USAGE:
+#
+# NOTES:
+#
+# LICENSE:
+#   Jordan Beauchamp beauchjord@gmail.com
+#   Released under the same terms as Sensu (the MIT license); see LICENSE
+#   for details.
+#
+
+require 'sensu-plugin/check/cli'
+require 'socket'
+
+#
+# Sensors
+#
+class CheckTemperature < Sensu::Plugin::Check::CLI
+  #
+  # Setup variables
+  #
+  def initialize
+    super
+    @crit_temp = []
+    @warn_temp = []
+  end
+
+  #
+  # Generate output
+  #
+
+  def usage_summary
+    (@crit_temp + @warn_temp).join(', ')
+  end
+
+  #
+  # Main function
+  #
+  def run
+    raw = `sensors`
+    sections = raw.split("\n\n")
+    metrics = {}
+    sections.each do |section|
+      section.split("\n").drop(1).each do |line|
+        begin
+          key, value = line.split(':')
+          key = key.downcase.gsub(/\s/, '')
+          if key[0..3] == 'temp' || key[0..3] == 'core'
```

Comment:
You could also include `physical` but I don't know enough about the workings of `sensors` to know for sure.